lua/typescript-tools/tsserver_provider.lua

```diff
@@ -25,10 +25,7 @@ function TsserverProvider.new()
   assert(util.bufname_valid(bufname), "Invalid buffer name!")
 
   self.root_dir = Path:new(config.get_root_dir(util.path.sanitize(bufname), bufnr))
-  self.npm_global_path = Path
-    :new(vim.fn.system([[node -p "process.execPath"]]):match "^%s*(.-)%s*$")
-    :parent()
-    :joinpath("..", "lib")
+  self.npm_global_path = Path:new(vim.fn.system "npm root -g"):parent()
```

Comment:
I'm wondering if it will work with other package managers like `yarn` or `pnpm`? Because I think that previous solution would actually work with all managers if I'm not mistaken

Reply:
This one resolves the NODE_PATH correctly, other package managers would install to the same location as `npm root -g`, as that's where node looks for the modules at runtime.

Reply:
The one downside of this approach is that if one has changed the defaults of where any of these package managers install things, everything breaks. In my case, yarn installs packages to `yarn global dir = ~/.local/share/yarn/global/node_modules` and tsserver cannot be found.

One solution would be to individually find `self.npm_global_path` according to a specified package manager, defaulting to npm.

Do you think this is a problem worth solving?